presto-accumulo/src/main/java/com/facebook/presto/accumulo/AccumuloClient.java

```diff
@@ -0,0 +1,1140 @@
+                            oldTable.getIndexTableName());
+                }
+                catch (Exception e1) {
+                    throw new PrestoException(INTERNAL_ERROR,
+                            "Failed to rollback table rename", e1);
+                }
+
+                throw new PrestoException(INTERNAL_ERROR,
+                        "Failed to rename index table", e);
+            }
+        }
+
+        try {
+            // We'll then create the metadata
+            metaManager.createTableMetadata(newTable);
+        }
+        catch (Exception e) {
+            // catch all to rollback the operation -- rename the table back
+            try {
+                conn.tableOperations().rename(newTable.getFullTableName(),
+                        oldTable.getFullTableName());
+
+                if (oldTable.isIndexed()) {
+                    conn.tableOperations().rename(newTable.getIndexTableName(),
+                            oldTable.getIndexTableName());
+                    conn.tableOperations().rename(newTable.getMetricsTableName(),
+                            oldTable.getMetricsTableName());
+                }
+            }
+            catch (Exception e1) {
+                throw new PrestoException(INTERNAL_ERROR,
+                        "Failed to rename table", e1);
+            }
+
+            throw new PrestoException(INTERNAL_ERROR, "Failed to create metadata",
+                    e);
+        }
+
+        try {
+            // We'll then delete the metadata
+            metaManager.deleteTableMetadata(oldName);
+        }
+        catch (Exception e) {
+            try {
+                // catch all to rollback the operation -- rename the table back and create the old
+                // metadata
+                conn.tableOperations().rename(newTable.getFullTableName(),
+                        oldTable.getFullTableName());
+
+                if (oldTable.isIndexed()) {
+                    conn.tableOperations().rename(newTable.getIndexTableName(),
+                            oldTable.getIndexTableName());
+                    conn.tableOperations().rename(newTable.getMetricsTableName(),
+                            oldTable.getMetricsTableName());
+                }
+
+                metaManager.createTableMetadata(oldTable);
+            }
+            catch (Exception e1) {
+                throw new PrestoException(INTERNAL_ERROR,
+                        "Failed to rollback rename table operation ", e1);
+            }
+
+            throw new PrestoException(INTERNAL_ERROR, "Failed to delete metadata",
+                    e);
+        }
+    }
+
+    /**
+     * Creates a new view with the given name and data.  Replaces a view if 'replace' is true.
+     *
+     * @param viewName Name of the view
+     * @param viewData Data for the view
+     * @param replace True to replace, false otherwise
+     */
+    public void createView(SchemaTableName viewName, String viewData, boolean replace)
```

Comment:
Wouldn't it be better to have two methods:
`createOrReplaceView` and `createView` instead of passing this `bool` flag?